### PR TITLE
:sheep: switched from npm to yarn

### PR DESCRIPTION
**Pull request by markysoft**
switched to yarn, upgraded node to 7.9, added start/test scripts

### CODE REVIEW

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder-pr-84 in nhsuk-dev succeeded (http://gp-finder-pr-84.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder-pr-84 in nhsuk-dev succeeded (http://gp-finder-pr-84.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder-pr-84 in nhsuk-dev succeeded (http://gp-finder-pr-84.dev.c2s.nhschoices.net)

File: Dockerfile [hunk 1]
```diff
@@ -13,8 +13,9 @@ WORKDIR /code
 ARG NODE_ENV=production
 ENV NODE_ENV=${NODE_ENV}
 
-COPY npm-shrinkwrap.json /code
-RUN if [ "$NODE_ENV" == "production" ]; then npm install --quiet --only=prod; else npm install --quiet ; fi
+COPY yarn.lock /code
+COPY package.json /code
+RUN yarn && if [ "$NODE_ENV" == "production" ]; then yarn install --production; else yarn install ; fi
```
Comment (beseven) [inline Dockerfile]: What does `yarn && yarn install` do compared to `yarn install` for example?

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder-pr-84 in nhsuk-dev succeeded (http://gp-finder-pr-84.dev.c2s.nhschoices.net)

File: README.md [hunk 2]
```diff
@@ -30,3 +30,6 @@ When finished with the application `docker-compose down -v` should be run to shu
 
 It is good practice to run the tests in a Docker container. To do this run `docker-compose -p gp-finder-test -f docker-compose-tests.yml up --build --force-recreate tests`. A new container will be started where the tests will run and rerun when changes are made to the source code.
 When finished with the tests run `docker-compose -p gp-finder-test -f docker-compose-tests.yml down -v` to tidy up.
+
+The script `scripts/start.sh` starts the application containers cleanly by running a docker-compose down, immediately followed by a docker-compose up.
```
Comment (beseven) [inline README.md]: Do we need to add something about running the app/tests in the form of `./scripts/start.sh` & `./scripts/test.sh`. Happy for this to be ignored if too much handholding in the readme.

File: scripts/start.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+docker-compose down -v
+docker-compose up --build --force-recreate
```
Comment (st3v3nhunt) [inline scripts/start.sh]: New line please

File: scripts/test.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+docker-compose -p gp-finder-test -f docker-compose-tests.yml down -v
+docker-compose -p gp-finder-test -f docker-compose-tests.yml up --build --force-recreate tests
```
Comment (st3v3nhunt) [inline scripts/test.sh]: New line please

**Comment by st3v3nhunt** [comment]
@markysoft A couple of additional comments on the PR that I didn't get in before @beseven merged it...

File: README.md (hunk 2 above)
Comment (beseven) [inline README.md]: @st3v3nhunt I'll add the changes, what do you think of the above?

**Pull request merged**